src/component/useFileUpload.js

```diff
@@ -110,37 +137,20 @@ export function useFileUpload({
     [],
   )
 
-  const applyUserProvidedSlices = useCallback(
-    files => {
-      let filePayload = []
-      if (multiple) {
-        if (filecountlimit) {
-          filePayload = [files.slice(0, filecountlimit)]
-        } else {
-          filePayload = [files]
-        }
-      } else {
-        filePayload = [files[0]]
-      }
-      return filePayload
-    },
-    [multiple, filecountlimit],
-  )
-
   const onDropEvent = useCallback(
     event => {
       const {dataTransfer} = event
       if (dataTransfer?.items?.length) {
         simpleAction(actionTypes.FILE_DROPPED, dispatch)
-        handleFiles(applyUserProvidedSlices(dataTransfer.files))
+        handleFiles(validateFileInput([...dataTransfer.files]))
```

Comment:
Didnt get the use of this, since dataTransfer.files is actully array and the above also creates an array!

Reply:
Actually, if you are directly passing the dataTransfer.files, unable to iterate it

Reply:
I think  due to some other issues, but it should work.